sklearn/utils/validation.py

```diff
@@ -37,6 +37,10 @@ class NotFittedError(ValueError, AttributeError):
     """
 
 
+class SingleDimensionWarning(UserWarning):
```

Comment:
If we want to define a new warning (not sure about it), it should probably inherit from ``DeprecationWarning``